File: tests/rosetta/transpiler/FS/case-sensitivity-of-identifiers.fs
// Generated 2025-07-27 23:45 +0700

exception Return

let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec main () =
    let mutable __ret : unit = Unchecked.defaultof<unit>
    try
        let __bench_start = _now()
        let __mem_start = System.GC.GetTotalMemory(true)
        let mutable pkg_dog: string = "Salt"
        let mutable Dog: string = "Pepper"
        let mutable pkg_DOG: string = "Mustard"
        let rec packageSees (d1: string) (d2: string) (d3: string) =
            let mutable __ret : Map<string, bool> = Unchecked.defaultof<Map<string, bool>>
            let mutable d1 = d1
            let mutable d2 = d2
            let mutable d3 = d3
            try
                printfn "%s" ((((("Package sees: " + d1) + " ") + d2) + " ") + d3)
                __ret <- Map.ofList [("pkg_dog", true); ("Dog", true); ("pkg_DOG", true)]
                raise Return
                __ret
            with
                | Return -> __ret
        let mutable d: Map<string, bool> = packageSees pkg_dog Dog pkg_DOG
        printfn "%s" (("There are " + (string (Seq.length d))) + " dogs.\n")
        let mutable dog: string = "Benjamin"
        d <- packageSees pkg_dog Dog pkg_DOG
        printfn "%s" ((((("Main sees:   " + dog) + " ") + Dog) + " ") + pkg_DOG)
        d <- Map.add "dog" true d
        d <- Map.add "Dog" true d
        d <- Map.add "pkg_DOG" true d
        printfn "%s" (("There are " + (string (Seq.length d))) + " dogs.\n")
        Dog <- "Samba"
        d <- packageSees pkg_dog Dog pkg_DOG
        printfn "%s" ((((("Main sees:   " + dog) + " ") + Dog) + " ") + pkg_DOG)
        d <- Map.add "dog" true d
        d <- Map.add "Dog" true d
        d <- Map.add "pkg_DOG" true d
        printfn "%s" (("There are " + (string (Seq.length d))) + " dogs.\n")
        let mutable DOG: string = "Bernie"
        d <- packageSees pkg_dog Dog pkg_DOG
        printfn "%s" ((((("Main sees:   " + dog) + " ") + Dog) + " ") + DOG)
        d <- Map.add "dog" true d
        d <- Map.add "Dog" true d
        d <- Map.add "pkg_DOG" true d
        d <- Map.add "DOG" true d
        printfn "%s" (("There are " + (string (Seq.length d))) + " dogs.")
        let __bench_end = _now()
        let __mem_end = System.GC.GetTotalMemory(true)
        printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

        __ret
    with
        | Return -> __ret
main()

File: tests/rosetta/transpiler/Kotlin/case-sensitivity-of-identifiers.kt
var _nowSeed = 0L
var _nowSeeded = false
fun _now(): Long {
    if (!_nowSeeded) {
        System.getenv("MOCHI_NOW_SEED")?.toLongOrNull()?.let {
            _nowSeed = it
            _nowSeeded = true
        }
    }
    return if (_nowSeeded) {
        _nowSeed = (_nowSeed * 1664525 + 1013904223) % 2147483647
        kotlin.math.abs(_nowSeed)
    } else {
        kotlin.math.abs(System.nanoTime())
    }
}

fun toJson(v: Any?): String = when (v) {
    null -> "null"
    is String -> "\"" + v.replace("\"", "\\\"") + "\""
    is Boolean, is Number -> v.toString()
    is Map<*, *> -> v.entries.joinToString(prefix = "{", postfix = "}") { toJson(it.key.toString()) + ":" + toJson(it.value) }
    is Iterable<*> -> v.joinToString(prefix = "[", postfix = "]") { toJson(it) }
    else -> toJson(v.toString())
}

fun user_main(): Unit {
    var pkg_dog: String = "Salt"
    var Dog: String = "Pepper"
    var pkg_DOG: String = "Mustard"
    fun packageSees(d1: String, d2: String, d3: String): MutableMap<String, Boolean> {
        println((((("Package sees: " + d1) + " ") + d2) + " ") + d3)
        return mutableMapOf<String, Boolean>("pkg_dog" to (true), "Dog" to (true), "pkg_DOG" to (true))
    }

    var d: MutableMap<String, Boolean> = packageSees(pkg_dog, Dog, pkg_DOG)
    println(("There are " + d.size.toString()) + " dogs.\n")
    var dog: String = "Benjamin"
    d = (packageSees(pkg_dog, Dog, pkg_DOG)) as MutableMap<String, Boolean>
    println((((("Main sees:   " + dog) + " ") + Dog) + " ") + pkg_DOG)
    (d)["dog"] = true
    (d)["Dog"] = true
    (d)["pkg_DOG"] = true
    println(("There are " + d.size.toString()) + " dogs.\n")
    Dog = "Samba"
    d = (packageSees(pkg_dog, Dog, pkg_DOG)) as MutableMap<String, Boolean>
    println((((("Main sees:   " + dog) + " ") + Dog) + " ") + pkg_DOG)
    (d)["dog"] = true
    (d)["Dog"] = true
    (d)["pkg_DOG"] = true
    println(("There are " + d.size.toString()) + " dogs.\n")
    var DOG: String = "Bernie"
    d = (packageSees(pkg_dog, Dog, pkg_DOG)) as MutableMap<String, Boolean>
    println((((("Main sees:   " + dog) + " ") + Dog) + " ") + DOG)
    (d)["dog"] = true
    (d)["Dog"] = true
    (d)["pkg_DOG"] = true
    (d)["DOG"] = true
    println(("There are " + d.size.toString()) + " dogs.")
}

fun main() {
    run {
        System.gc()
        val _startMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
        val _start = _now()
        user_main()
        System.gc()
        val _end = _now()
        val _endMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
        val _durationUs = (_end - _start) / 1000
        val _memDiff = kotlin.math.abs(_endMem - _startMem)
        val _res = mapOf("duration_us" to _durationUs, "memory_bytes" to _memDiff, "name" to "main")
        println(toJson(_res))
    }
}
